### PR TITLE
Applications automatically restart to pick up new configuration

### PR DESCRIPTION
**Pull request by alexwlchan**
Replaces #61.

This adds a new Lambda that watches the config files in S3. When one of them is changed, it stops all the running instances of that task, forcing a restart so that they pick up the new configuration.

### CODE REVIEW

File: lambdas/stop_running_tasks.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def identify_cluster_by_app_name(ecs, app_name):
+    """
+    Given the name of one of our applications (e.g. api, calm_adapter),
+    return the ARN of the cluster the task runs on.
+    """
+    for cluster in ecs.list_clusters()['clusterArns']:
+        for serviceArn in ecs.list_services(cluster=cluster)['serviceArns']:
+            _, serviceName = serviceArn.split('/')
```
Comment (alexwlchan) [inline lambdas/stop_running_tasks.py]: add an example service ARN

Comment (alexwlchan) [inline lambdas/stop_running_tasks.py]: dash/underscore

File: lambdas/stop_running_tasks.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            if serviceName == app_name:
+                return cluster
```
Comment (alexwlchan) [inline lambdas/stop_running_tasks.py]: throw an error if we don't find the cluster

File: terraform/lambda/role.tf [hunk 3]
```diff
@@ -26,16 +26,7 @@ data "aws_iam_policy_document" "cloudwatch_logs" {
   statement {
     actions = [
       "logs:CreateLogGroup",
```
Comment (alexwlchan) [inline terraform/lambda/role.tf]: take this out

**Comment by kenoir** [comment]
This is pretty awesome.

**Pull request merged**